src/no_std/node.rs

```diff
@@ -12,8 +13,10 @@ use alloc::boxed::Box;
 use core::num::NonZeroUsize;
 use core::ptr;
 
+pub(super) type GenericTags<T> = Box<dyn FnMut() -> T + Send + Sync + 'static>;
```

Comment:
is it really a good idea to require an allocation for usage? (why not make it generic in such a way that T is a function itself, which would be more general and probably also faster)

Reply:
Unfortunately it's necessary. This structure is stored directly inside of `Queue<T>`, so we can't use the function as a parameter without it propagating all the way to `Event` and `EventListener`.

In addition, this is on a cold path, so it's unlikely to happen in the first place.

Reply:
hm so why can't we parameterize `Event` and `EventListener` with that anyways and create type aliases for this case? It would blow up the types a bit, but I don't think it would be bad in practical application

Reply:
The closure stored in these boxes isn't a user provided closure; it's [this closure](https://github.com/smol-rs/event-listener/blob/a8c3aa76c2d4aee7f8dd0fa9c2098e753113a917/src/no_std.rs#L140). Until we have TAIT, we can't actually get a type for this closure, therefore we have to box it for now.

Reply:
Ohhh... but why do we have to use such a closure and not package it into a custom struct and idk... have it implement `Iterator` or such (or why not just a counter and a generating function?)

Reply:
You're right. I've replaced the boxed closure with a `Vec` of tags. It should have the same effect.